src/common/repo/git-storage.ts

```diff
@@ -0,0 +1,61 @@
+import type { Repo } from ".";
+
+export class GitStorage {
+  private repo: Repo;
+  private cache: object = {};
+
+  constructor(repo: Repo){
+    this.repo = repo;
+    this.cache['branch'] = {};
+    this.cache['files'] = {};
+  }
+  
+  public async createBranch(name: string, treeSha: string): Promise<string> {
+    if (this.cache['branch'][name]) {
+      return this.cache['branch'][name];
+    }
+    
+    let  existingBranch = null
+    await this.repo.api.rest.git.getRef({
+      owner: this.repo.owner, repo: this.repo.name, ref: `heads/${name}`
+    }).then(response => existingBranch = response.data.object.sha)
+    .catch(_ => console.log("Branch: ", `heads/${name}`, " does not exist"));
+
+    if(existingBranch){
+      return this.cache['branch'][name] = existingBranch.data.object.sha;
+    } else {
```

Comment:
To be consistent with the rest of the code, remove this else after an early return.